autoPyTorch/pipeline/components/training/data_loader/base_data_loader.py

```diff
@@ -45,10 +45,8 @@ def __init__(self, batch_size: int = 64) -> None:
 
         # Define fit requirements
         self.add_fit_requirements([
-            FitRequirement("dataset", (str,), user_defined=True, dataset_property=True),
-            FitRequirement("root", (str,), user_defined=True, dataset_property=True),
-            FitRequirement("split_id", (int,), user_defined=True, dataset_property=True),
-            FitRequirement("train_indices", (List[int],), user_defined=True, dataset_property=True),
+            FitRequirement("split_id", (int,), user_defined=True, dataset_property=False),
```

Comment:
Again

Reply:
Also for split id, they will again be calculated based on the fold that is being executed in case of cross validation or 0 in case of holdout. As a single dataset could be used for multiple runs of the pipeline and therefore, it should only contain the splits calculated at the beginning and the split id should be handled in the API or the AutoML class which can give different values for different runs of the pipeline in the same experiment

Reply:
Yes, I see the comment above 👍